test/InstantSearch/lifecycle.js

```diff
@@ -100,3 +105,71 @@ test('InstantSearch: lifecycle', function(t) {
 
   t.end();
 });
+
+test('InstantSearch: lifecycle, the widget with __initLast set to true should be called last.', function(t) {
+  var EventEmitter = require('events').EventEmitter;
+
+  var proxyquire = require('proxyquire');
+  var sinon = require('sinon');
+
+  // simple mocks
+  var client = {algolia: 'client'};
+  var helper = new EventEmitter();
+  helper.search = sinon.spy();
+  helper.state = 'state';
+
+  // hijack `algoliasearch` and `algoliasearchHelper` modules
+  var algoliasearch = sinon.stub().returns(client);
+  var algoliasearchHelper = sinon.stub().returns(helper);
+
+  var appId = 'appId';
+  var apiKey = 'apiKey';
+  var indexName = 'lifeycle';
+
+  var searchParameters = {some: 'configuration', another: {config: 'parameter'}};
+
+  var lastWidget;
+  var nbConfiguredWidget = 0;
+  var widgets = _(_.range(10)).map(function(i) {
+    var widget = {
+      getConfiguration: function() {
+        nbConfiguredWidget++;
+        t.ok(
+          lastWidget.getConfiguration.notCalled,
+          'The configuration of the last widget should not have been called yet'
+        );
+      },
+      init: sinon.spy(),
+      render: sinon.spy()
+    };
+    if (i === 5) {
+      widget.__initLast = true;
+      widget.getConfiguration = function() {
+        widget.getConfiguration.notCalled = false;
+        t.equals(nbConfiguredWidget, 9);
+        t.end();
+      };
+      widget.getConfiguration.notCalled = true;
+      lastWidget = widget;
+    }
+    return widget;
+  });
+
+  var InstantSearch = proxyquire('../../lib/InstantSearch', {
+    algoliasearch: algoliasearch,
+    'algoliasearch-helper': algoliasearchHelper
+  });
+
+  var search = new InstantSearch({
+    appId: appId,
+    apiKey: apiKey,
+    indexName: indexName,
+    searchParameters: searchParameters
+  });
+
+  widgets.forEach(function(w) {
+    search.addWidget(w);
+  }).value();
```

Comment:
AHAH :) Lodash implements lazy operations when you wrap something. In order for it to start any kind of computation, you have to manually get the value.